examples/RNOneSignal/App.js

```diff
@@ -66,80 +66,81 @@ export default class App extends Component {
-        OneSignal.inFocusDisplaying(2);
-
-        // If the app requires privacy consent check if it has been set yet
+        //If the app requires privacy consent check if it has been set yet
         if (requiresPrivacyConsent) {
         // async 'then' is only so I can sleep using the Promise helper method
             OneSignal.userProvidedPrivacyConsent().then(async (granted) => {
                 // For UI testing purposes wait X seconds to see the loading state
                 await sleep(0);
 
-                console.log('Privacy Consent status: ' + granted);
+                console.log('OneSignal: Privacy Consent status: ' + granted);
                 this.setState({hasPrivacyConsent:granted, isPrivacyConsentLoading:false});
             });
         }
 
-        OneSignal.getPermissionSubscriptionState((response) => {
-            console.log('Device state:');
-            console.log(response);
-
-            let notificationsEnabled = response['notificationsEnabled'];
-            let isSubscribed = response['subscriptionEnabled'];
-
-            this.setState({isSubscribed:notificationsEnabled && isSubscribed, isSubscriptionLoading:false}, () => {
-                OneSignal.setSubscription(isSubscribed);
-            });
-        });
+        let deviceState = await OneSignal.getDeviceState();
+        this.setState({ isSubscribed: deviceState.isSubscribed });
 
         // Examples for using native IAM public methods
-//        this.oneSignalInAppMessagingExamples();
+        // this.oneSignalInAppMessagingExamples();
 
         // Examples for using native Outcome Event public methods
-//        this.oneSignalOutcomeEventsExamples();
-
-    }
+        // this.oneSignalOutcomeEventsExamples();
 
-    async componentDidMount() {
-        OneSignal.addEventListener('received', this.onNotificationReceived);
-        OneSignal.addEventListener('opened', this.onNotificationOpened);
-        OneSignal.addEventListener('ids', this.onIdsAvailable);
-//        OneSignal.addEventListener('subscription', this.onSubscriptionChange);
-//        OneSignal.addEventListener('permission', this.onPermissionChange);
-        OneSignal.addEventListener('emailSubscription', this.onEmailSubscriptionChange);
-        OneSignal.addEventListener('inAppMessageClicked', this.onInAppMessageClicked);
     }
 
     componentWillUnmount() {
-        OneSignal.removeEventListener('received', this.onNotificationReceived);
-        OneSignal.removeEventListener('opened', this.onNotificationOpened);
-        OneSignal.removeEventListener('ids', this.onIdsAvailable);
-//        OneSignal.removeEventListener('subscription', this.onSubscriptionChange);
-//        OneSignal.removeEventListener('permission', this.onPermissionChange);
-        OneSignal.removeEventListener('emailSubscription', this.onEmailSubscriptionChange);
-        OneSignal.removeEventListener('inAppMessageClicked', this.onInAppMessageClicked);
+        OneSignal.clearHandlers();
```

Comment:
Wondering in testing if we observed if still need to clean up handlers?

Reply:
There's two things the event manager tracks: 1) the RN listeners & 2) the JS side handlers stored in a Map object

When I tried logging the contents of the listeners array, there was nothing in there for some reason. I found this very strange, especially given this example:
```js
import { NativeEventEmitter, NativeModules } from 'react-native';
...

  componentDidMount() {
    ...
    const eventEmitter = new NativeEventEmitter(NativeModules.ToastExample);
    this.eventListener = eventEmitter.addListener('EventReminder', (event) => {
       console.log(event.eventProperty) // "someValue"
    });
    ...
  }

  componentWillUnmount() {
    this.eventListener.remove(); //Removes the listener
  }
```
in the [official docs](https://reactnative.dev/docs/native-modules-android#sending-events-to-javascript).

That said, I did notice for the handlers there can be multiple instances of (i.e. the adders), it seems that these _are_ stored somewhere in memory even after the component re-mounts. Note that I don't know that this would impact a production environment, but nevertheless this would lead to confusion in a testing environment where saving the project auto-reloads the app via the metro-bundler. 

I added the `clearHandlers` function for use in the top-level component's `componentWillUnmount`